sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -0,0 +1,735 @@
+            /// Checks for and returns any bindings found in the Return Type of the method
+            /// </summary>
+            private bool TryGetReturnTypeBindings(MethodDeclarationSyntax method, SemanticModel model, bool hasHttpTrigger, bool hasOutputBinding, out IList<IDictionary<string, string>>? bindingsList)
+            {
+                TypeSyntax returnTypeSyntax = method.ReturnType;
+                ITypeSymbol? returnTypeSymbol = model.GetSymbolInfo(returnTypeSyntax).Symbol as ITypeSymbol;
+                bindingsList = new List<IDictionary<string, string>>();
+
+                if (returnTypeSymbol is null)
+                {
+                    _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.SymbolNotFound, returnTypeSyntax.GetLocation(), nameof(returnTypeSymbol)));
+                    bindingsList = null;
+                    return false;
+                }
+
+                if (returnTypeSymbol != null &&
+                    !SymbolEqualityComparer.Default.Equals(returnTypeSymbol, Compilation.GetTypeByMetadataName(Constants.VoidType)) &&
+                    !SymbolEqualityComparer.Default.Equals(returnTypeSymbol, Compilation.GetTypeByMetadataName(Constants.TaskType)))
+                {
+                    // If there is a Task<T> return type, inspect T, the inner type.
+                    if (SymbolEqualityComparer.Default.Equals(returnTypeSymbol, Compilation.GetTypeByMetadataName(Constants.TaskGenericType))) 
+                    {
+                        GenericNameSyntax genericSyntax = (GenericNameSyntax)returnTypeSyntax;
+                        var innerTypeSyntax = genericSyntax.TypeArgumentList.Arguments.First(); // Generic task should only have one type argument
+                        returnTypeSymbol = model.GetSymbolInfo(innerTypeSyntax).Symbol as ITypeSymbol;
+
+                        if (returnTypeSymbol is null)
+                        {
+                            _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.SymbolNotFound, genericSyntax.Identifier.GetLocation(), nameof(returnTypeSymbol)));
+                            bindingsList = null;
+                            return false;
+                        }
+                    }
+
+                    if (SymbolEqualityComparer.Default.Equals(returnTypeSymbol, Compilation.GetTypeByMetadataName(Constants.HttpResponseType))) // If return type is HttpResponseData
+                    {
+                        bindingsList.Add(GetHttpReturnBinding(Constants.ReturnBindingName));
+                        hasOutputBinding = true;
+                    }
+                    else
+                    {
+                        var members = returnTypeSymbol.GetMembers();
+                        var foundHttpOutput = false;
+
+                        foreach (var m in members)
+                        {
+                            // Check if this attribute is an HttpResponseData type attribute
```

Comment:
Should we skip non property type members ?

    if (m is not IPropertySymbol memberProperty)
    {
        continue;
    }
    // use memberProperty below as neeeded.
    
Or even add a where clause to the `GetMembers` result above.

    returnTypeSymbol.GetMembers().Where(a => a is IPropertySymbol);